http4k-format/core/src/main/kotlin/org/http4k/format/Json.kt

```diff
@@ -88,8 +93,51 @@ interface Json<NODE> {
     fun textValueOf(node: NODE, name: String): String?
 
     operator fun <T> invoke(fn: Json<NODE>.() -> T): T = run(fn)
+
+    // Utility methods for mapping to objects - used when we don't know which implementation we are using
+    fun <T : Any> asA(mapper: (NODE) -> T): (NODE) -> T = { mapper(it.requireObject()) }
+    fun <T : Any> asArray(mapper: (NODE) -> T): (NODE) -> List<T> = { elements(it.requireArray()).map(asA(mapper)) }
+    fun field(): BiDiLensSpec<NODE, String> = fieldLens(StringParam) { name, node -> textValueOf(node, name) }
+    fun <T : Any> obj(mapper: (NODE) -> T): BiDiLensSpec<NODE, T> = fieldLens(ObjectParam) { _, node -> mapper(node) }
+    fun <T : Any> array(itemType: ParamMeta, mapper: (NODE) -> T): BiDiLensSpec<NODE, List<T>> =
+        fieldLens(ArrayParam(itemType)) { _, node -> elements(node).map(mapper) }
+    // ---
+
+    private fun <T : Any> fieldLens(paramMeta: ParamMeta, mapper: (String, NODE) -> T?): BiDiLensSpec<NODE, T> =
+        BiDiLensSpec("field", paramMeta,
```

Comment:
@FredNordin do you need to introduce Lens at the field level to start with?

Unless the failure messages are better than what you usually get in the Body lens, I'd suggest adding this in a future iteration. Otherwise, I'm afraid we're adding something that will make debugging even harder than at the moment. /cc @daviddenton

Reply:
Exactly the kind of feedback I was hoping for - thanks @s4nchez !
I can change it around to not use lenses and see what it looks like. I totally get your comment to try to keep it as "json" pure as possible and not involve lenses here, but maybe add them elsewhere, e.g. in the `org.http4k.lens` package.

Reply:
If lenses are even needed for this, obviously...

Reply:
> maybe add them elsewhere, e.g. in the org.http4k.lens package
If that's doable, I'd try that.

Also, I suggest including tests for failure scenarios, so we can get a feel for the kind of user feedback people could get in those cases.

Reply:
Comprehensive set of tests will be added as soon as we're agreed on the approach and implementation.